### PR TITLE
refactor: get last active deployments using deployments table

### PR DESCRIPTION
**Pull request by marianogoldman**
Second part of https://github.com/decentraland/catalyst/issues/1112.

This PR reimplements the code for `getLastActiveDeploymentsOnPointers()` by running a query on the deployments table, and does not use `last_deployed_pointers` table anymore. `last_deployed_pointers` will soon be deleted. Class `LastDeployedPointersRepository` is deleted.

It also removes `PointerHistoryRepository` as the content was no longer used since #1156.

### CODE REVIEW

**Review by pedrotambo — APPROVED** [review]

File: content/src/service/pointers/PointerManager.ts [hunk 1]
```diff
@@ -12,72 +9,38 @@ export class PointerManager {
    * Commit a new entity, and return the 'before & after' generated by the deployment
    */
   async referenceEntityFromPointers(
-    lastDeployedPointersRepo: LastDeployedPointersRepository,
+    deploymentsRepo: DeploymentsRepository,
     deploymentId: DeploymentId,
-    entity: Entity
+    entity: Entity,
+    overwrote: Set<number>,
+    overwritten: boolean
```
Comment (agusaldasoro) [inline content/src/service/pointers/PointerManager.ts:16]: what is this parameter used for? How is it calculated?

File: content/src/service/pointers/PointerManager.ts [hunk 2]
```diff
@@ -12,72 +9,38 @@ export class PointerManager {
...
-    // Prepare variables
     const result: DeploymentResult = new Map()
-    const overwrite: Set<string> = new Set()
 
-    lastDeployments.forEach((lastDeployment) => {
-      // Calculate the intersection of pointers between the last deployment and the new one
-      const intersection: Set<string> = intersect(lastDeployment.pointers, entity.pointers)
+    if (!overwritten) {
+      // Fetch overwritten deployments from the DB
+      const overwrittenDeployments = await deploymentsRepo.getDeployments(overwrote)
```
Comment (agusaldasoro) [inline content/src/service/pointers/PointerManager.ts:22]: Is it correct to name this `overwritten` if we are querying by `overwrote`?

File: content/src/service/pointers/PointerManager.ts [hunk 3]
```diff
@@ -12,72 +9,38 @@ export class PointerManager {
    * Commit a new entity, and return the 'before & after' generated by the deployment
    */
   async referenceEntityFromPointers(
```
Comment (agusaldasoro) [inline content/src/service/pointers/PointerManager.ts:11]: Can we add some more comments here explaining what does this method do in general and in each line?

Comment (agusaldasoro) [reply content/src/service/pointers/PointerManager.ts:11]: Does this endpoint make an update in the db or is just a read only?

Comment (marianogoldman) [reply content/src/service/pointers/PointerManager.ts:11]: It's read-only

File: content/src/service/pointers/PointerManager.ts (hunk 2 above)
Comment (marianogoldman) [reply content/src/service/pointers/PointerManager.ts]: Renamed and documented

**Pull request merged**